### PR TITLE
Add backend support and tests for copying all artifact sources to files

### PR DESCRIPTION
**Pull request by balzer814**
Signed-off-by: Lukas Balzer <balzer814_dev@web.de>

Move the method to copy the source files of a artifact template to the backend
Add tests for copy method

- [x] Ensure that the commit message is [a good commit message](https://github.com/erlang/otp/wiki/Writing-good-commit-messages)
- [ ] Change in CHANGELOG.md described
- [x] Tests created for changes
- [ ] Screenshots added (for UI changes)

### CODE REVIEW

**Comment by genie-winery** [comment]
Can one of the admins verify this patch?

File: org.eclipse.winery.repository.rest/src/main/java/org/eclipse/winery/repository/rest/resources/entitytemplates/artifacttemplates/ArtifactTemplateResource.java [hunk 1]
```diff
@@ -86,6 +92,30 @@ public void synchronizeReferences() throws IOException {
 		BackendUtils.synchronizeReferences((ArtifactTemplateId) this.id);
 	}
 
+	@POST
+	@Consumes(MediaType.APPLICATION_JSON)
+	public Response copySourceToFilesResource(ArtifactResourcesApiData data) {
```
Comment (koppor) [inline org.eclipse.winery.repository.rest/src/main/java/org/eclipse/winery/repository/rest/resources/entitytemplates/artifacttemplates/ArtifactTemplateResource.java]: Could you document that `data` is used to state that only a subset of files has to be copied? Use the [@ApiParam](https://github.com/swagger-api/swagger-core/wiki/Annotations-1.5.X#apiparam) value.

File: org.eclipse.winery.repository.rest/src/main/java/org/eclipse/winery/repository/rest/resources/entitytemplates/artifacttemplates/ArtifactTemplateResource.java [hunk 2]
```diff
@@ -86,6 +92,30 @@ public void synchronizeReferences() throws IOException {
...
+		List<String> artifactList = data.getArtifactList();
+		ArtifactTemplateDirectoryId sourceDir = new ArtifactTemplateSourceDirectoryId((ArtifactTemplateId) this.id);
+		FilesResource filesResource = getFilesResource();
+		for (RepositoryFileReference ref : RepositoryFactory.getRepository().getContainedFiles(sourceDir)) {
+			if (artifactList == null || artifactList.contains(ref.getFileName())) {
+				try {
+					InputStream inputStream = RepositoryFactory.getRepository().newInputStream(ref);
+					String fileName = ref.getFileName();
+					String subDirectory = "";
```
Comment (koppor) [inline org.eclipse.winery.repository.rest/src/main/java/org/eclipse/winery/repository/rest/resources/entitytemplates/artifacttemplates/ArtifactTemplateResource.java]: Does following Java8 thing work?

   subDirectory = ref.getSubDirectory().map(s -> s.toString()).orElse("");

This is shorter than the currently programmed four lines.

File: org.eclipse.winery.repository.rest/src/main/java/org/eclipse/winery/repository/rest/resources/entitytemplates/artifacttemplates/ArtifactTemplateResource.java [hunk 3]
```diff
@@ -86,6 +92,30 @@ public void synchronizeReferences() throws IOException {
...
+					if (ref.getSubDirectory().isPresent()) {
+						subDirectory = ref.getSubDirectory().get().toString();
+					}
+					filesResource.putFile(fileName, subDirectory, inputStream);
+				} catch (IOException e) {
+					return Response.noContent().build();
```
Comment (koppor) [inline org.eclipse.winery.repository.rest/src/main/java/org/eclipse/winery/repository/rest/resources/entitytemplates/artifacttemplates/ArtifactTemplateResource.java]: Please `debug` log the exception. Further, return an internal server error. The user should be notified that an error happened.

File: org.eclipse.winery.repository.rest/src/main/java/org/eclipse/winery/repository/rest/resources/apiData/ArtifactResourcesApiData.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.eclipse.winery.repository.rest.resources.apiData;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class ArtifactResourcesApiData {
+	public String[] artifactNames;
```
Comment (koppor) [inline org.eclipse.winery.repository.rest/src/main/java/org/eclipse/winery/repository/rest/resources/apiData/ArtifactResourcesApiData.java]: Do not use these basic types in Java. Use List<String>.

**Pull request merged**